File: programming/labaratory-work-1/solutions/solution-7/main.c
#include <stdio.h>

int main() {
  int n = 0;

  scanf_s("%d", &n);

  int count = 0;
  int prev_k = 0;

  for (int i = 0; i < n; i++) {
    int k = 0;

    scanf_s("%d", &k);

    if (k > 0 && prev_k < 0 || k < 0 && prev_k > 0) {
      count++;
    }

    prev_k = k;
  }

  printf_s("%d", count);

  return 0;
}

File: programming/labaratory-work-1/solutions/solution-7/go/main.go
package main

import "fmt"

func main() {
	var n int

	fmt.Scanf("%d", &n)

	var count int
	var prevK int

	for i := 0; i < n; i++ {
		var k int

		fmt.Scanf("%d", &k)

		if k > 0 && prevK < 0 || k < 0 && prevK > 0 {
			count++
		}

		prevK = k
	}

	fmt.Printf("%d", count)
}